### PR TITLE
Rework Driver-registrar into two programs

### PR DESCRIPTION
**Pull request by lpabon**
This splits driver-registrar into: node-driver-registrar, and
cluster-driver-registrar programs. Each of these has their respective
containers and repos.

Signed-off-by: Luis Pabón <luis@portworx.com>

### CODE REVIEW

File: cmd/cluster-driver-registrar/main.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+			"- csi.storage.k8s.io/pod.name: pod.Name\n"+
+			"- csi.storage.k8s.io/pod.namespace: pod.Namespace\n"+
+			"- csi.storage.k8s.io/pod.uid: string(pod.UID)",
+	)
+	connectionTimeout = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
+	csiAddress        = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	showVersion       = flag.Bool("version", false, "Show version.")
+	version           = "unknown"
+	// List of supported versions
+	supportedVersions = []string{"0.2.0", "0.3.0"}
```
Comment (saad-ali) [inline cmd/cluster-driver-registrar/main.go]: Looks like this version of `main.go` is out of date. The deleted `cmd/driver-registrar/main.go` has the correct fields here.
Can you grab `cmd/driver-registrar/main.go` to pick up the latest changes and modify it.

**Comment by saad-ali** [comment]
Also with `cmd/driver-registrar/node_register.go` being deleted, see if there are any dependencies you can drop from `Gopkg.toml`

File: cmd/cluster-driver-registrar/main.go (hunk 1 above)
Comment (lpabon) [reply cmd/cluster-driver-registrar/main.go]: will do

File: Makefile [hunk 2]
```diff
@@ -12,10 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: all driver-registrar clean test
+.PHONY: all cluster-driver-registrar clean test
 
 REGISTRY_NAME=quay.io/k8scsi
-IMAGE_NAME=driver-registrar
+IMAGE_NAME=cluster-driver-registrar
```
Comment (saad-ali) [inline Makefile]: Change this to `IMAGE_NAME=csi-cluster-driver-registrar`

File: Makefile [hunk 3]
```diff
@@ -12,10 +12,10 @@
...
 IMAGE_VERSION=canary
```
Comment (saad-ali) [inline Makefile:19]: When we create a 1.0 branch make sure to update this.

**Comment by lpabon** [comment]
Minus 2Mil lines? wow

**Pull request merged**